File: scripts/config/llm_config.py
"""LLM configuration for CrewAI agents."""
from crewai import LLM
from config.settings import settings


def get_brain_llm() -> LLM:
    """Get LLM for Brain layer (Gemini 2.0 Flash)."""
    return LLM(
        model=f"gemini/gemini-2.0-flash",
        api_key=settings.google_api_key,
        temperature=0.7,
        max_tokens=4096
    )


def get_execution_llm() -> LLM:
    """Get LLM for Execution layer (Gemini)."""
    return LLM(
        model=f"gemini/gemini-2.0-flash",
        api_key=settings.google_api_key,
        temperature=0.8,
        max_tokens=2048
    )


def get_community_llm() -> LLM:
    """Get LLM for Community layer (Llama)."""
    return LLM(
        model=f"openai/{settings.llm.community_model}",
        api_key=settings.openai_api_key,
        base_url="https://api.llama-api.com/v1",  # Or local Llama server
        temperature=0.9,
        max_tokens=1024
    )

File: src/config/llmConfig.ts
/** LLM configuration for each layer. */
import { settings } from "./settings.js";

export interface LLMInstance {
  model: string;
  apiKey: string;
  baseUrl?: string;
  temperature: number;
  maxTokens: number;
}

export function getBrainLlm(): LLMInstance {
  return {
    model: "gemini/gemini-2.0-flash",
    apiKey: settings.googleApiKey,
    temperature: 0.7,
    maxTokens: 4096,
  };
}

export function getExecutionLlm(): LLMInstance {
  return {
    model: "gemini/gemini-2.0-flash",
    apiKey: settings.googleApiKey,
    temperature: 0.8,
    maxTokens: 2048,
  };
}

export function getCommunityLlm(): LLMInstance {
  return {
    model: `openai/${settings.llm.communityModel}`,
    apiKey: settings.openaiApiKey,
    baseUrl: "https://api.llama-api.com/v1",
    temperature: 0.9,
    maxTokens: 1024,
  };
}
